homeassistant/components/homekit_controller/connection.py

```diff
@@ -220,7 +222,7 @@ async def async_setup(self) -> None:
                 # BLE devices may sleep and we can't force a connection
                 raise
             entry.async_on_unload(
-                self.hass.bus.async_listen_once(
+                self.hass.bus.async_listen(
```

Comment:
This is intentional as you can't call the callable on async_listen_once once its fired to unsub it.

I'd remove the `entry.async_on_unload` instead but then it won't be canceled if you delete the config entry before startup which someone might do if its taking a while

fixes
```
Oct 24 03:18:01 homeassistant homeassistant[483]: 2022-10-23 22:18:01.747 ERROR (MainThread) [homeassistant.core] Unable to remove unknown job listener _FilterableJob(job=<Job HassJobType.Callback <function HKDevice._async_retry_populate_ble_accessory_state at 0xffff8da84550>>, event_filter=None, run_immediately=False)
Oct 24 03:18:01 homeassistant homeassistant[483]: Traceback (most recent call last):
Oct 24 03:18:01 homeassistant homeassistant[483]:   File "/usr/src/homeassistant/homeassistant/core.py", line 1069, in _async_remove_listener
Oct 24 03:18:01 homeassistant homeassistant[483]:     self._listeners[event_type].remove(filterable_job)
Oct 24 03:18:01 homeassistant homeassistant[483]: KeyError: 'homeassistant_started'
```